gusto/time_discretisation/time_discretisation.py

```diff
@@ -219,31 +226,33 @@ def setup(self, equation, apply_bcs=True, *active_labels):
 
             else:
                 if self.wrapper_name == "supg":
-                    self.wrapper.setup()
+                    for field_name in self.wrapper_field_names:
+                        self.wrapper.setup(field_name)
+                        new_test = self.wrapper.test
+                        self.residual = self.residual.label_map(
+                            lambda t: t.get(prognostic) == field_name and t.has_label(transport),
```

Comment:
can we change this so that we are testing whether the label is not `time_derivative`? I think it's better if the time discretisation to not know what terms it contains

Reply:
If we apply it to the momentum eq, then it will try to replace the test function for all terms with u in, which will be problematic..